### PR TITLE
Improve error message if filename does not match document/tracking/id and let it be reported by the proper reporter

### PR DESCRIPTION
**Pull request by JanHoefelmeyer**
Solves https://github.com/csaf-poc/csaf_distribution/issues/380

### CODE REVIEW

**Review by tschmidtb51 — APPROVED** [review]
LGTM and works

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
This might be unrelated (or fixed by #357) - but if I try this PR with an empty ROLIE feed. I get a memory access violation:

```
roliecheck.go:247
processor.go:977
processor.go:322
processor.go:243
main.go:151
main.go:174
```

Maybe something is not initialized yet?

**Comment by JanHoefelmeyer** [comment]
> This might be unrelated (or fixed by #357) - but if I try this PR with an empty ROLIE feed. I get a memory access violation:
> 
> ```
> roliecheck.go:247
> processor.go:977
> processor.go:322
> processor.go:243
> main.go:151
> main.go:174
> ```
> 
> Maybe something is not initialized yet?

The branch seems to be based off of an older version (1 commit ahead, 41 behind main) which still has that error not yet fixed. The changes within this PR do not actually change anything that should be able to break something elsewhere.

They only rephrase an error message and use a different reporter. To be on the safe side, I merged main into this branch, which - on my side at least - fixes this error.

File: cmd/csaf_checker/processor.go [hunk 1]
```diff
@@ -695,7 +695,7 @@ func (p *processor) integrity(
 		}
 
 		if err := util.IDMatchesFilename(p.expr, doc, filepath.Base(u)); err != nil {
-			p.invalidAdvisories.error("%s: %v\n", u, err)
+			p.badFilenames.error("%s: %v\n", u, err)
```
Comment (tschmidtb51) [inline cmd/csaf_checker/processor.go:698]:
```suggestion
                        p.badFilenames.error("%s: %v", u, err)

**Review by tschmidtb51 — DISMISSED** [review]
LGTM and works

**Comment by tschmidtb51** [comment]
@s-l-teichmann ready for review

**Review by tschmidtb51 — APPROVED** [review]
LGTM and works

**Review by s-l-teichmann — APPROVED** [review]
LGTM

**Pull request merged**